### DISCUSSION TITLE
MSTest SDK

### DISCUSSION THREAD

**Discussion by avivanoff** [microsoft/testfx#4065] [Q&A] [answered]
Is [MSTest.Sdk](https://learn.microsoft.com/en-us/dotnet/core/testing/unit-testing-mstest-sdk) a recommended approach for the new projects? Should I update my existing ones? Is .NET 9 required?

**Comment by Evangelink** [comment, marked as answer]
Hi @avivanoff,

Ideally this is the solution we would like to recommend but it is facing some limitations in the integration with NuGet which can turn off some the users (e.g. you cannot update the version through VS or dotnet/nuget CLIs), it also has a small performance overhead due to how NuGet/MSBuild resolution is working.

This being said, if you don't mind updating manually or using Dependabot, we would strongly advise to use it as it removes a lot of boilerplate from your projects and provides good defaults.

As for .NET version there is no requirement outside having SDK style project (i.e. you can use `MSTest.Sdk` for .NET 462+ without problem).
